pkg/packagekit/kpatch.jsx

```diff
@@ -211,25 +218,37 @@ export class KpatchSettings extends React.Component {
     }
 
     render() {
-        // Not yet recognized
-        if (this.state.loaded === false || this.state.patchName === null)
+        // don't show anything during initial detection
+        if ((this.state.loaded === false || this.state.patchName === null) && !this.state.showLoading)
             return null;
 
         // Not supported on this system
-        if (this.state.unavailable.length > 0)
+        if (this.state.unavailable.length > 0 && !this.state.showLoading)
             return null;
-
-        let state = _("Enabled");
+        let state;
         let actionText = _("Edit");
         let action = () => this.setState({ showModal: true });
 
-        if (this.state.missing.length > 0) {
+        if (this.state.loaded === false || this.state.patchName === null) {
+            // Not yet recognized
+            state = <Spinner isSVG size="md" />;
+        } else if (this.state.unavailable.length > 0) {
+            state = <Popover headerContent={ _("Missing packages") } bodyContent={ this.state.unavailable.join(", ") }>
```

Comment:
Not sure if I would say "missing" as that suggest they are not installed. If they are missing, we ofter installing them, they are "unavailable".

Reply:
Good point -- I'll adjust this after the os-release dependency PR lands.

Reply:
Changed to "Unavailable packages".